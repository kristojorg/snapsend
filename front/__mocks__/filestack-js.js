/* Mocking filestack-js */

export const FAILED_MESSAGE = 'Image Upload Failed';

export const successResult = {
  filename: 'img.jpg',
  handle: 'EL30NOJkQ9WJ2B0TBdoE',
  mimetype: 'image/jpeg',
  size: 1043074,
  status: 'Stored',
  url: 'https://cdn.filestackcontent.com/EL30NOJkQ9WJ2B0TBdoE',
};

export const errorResult = FAILED_MESSAGE;

export function rejectWithError(err = errorResult) {
  return new Promise((resolve, reject) => {
    return process.nextTick(() => reject(err));
  });
}

export function resolveSuccessfully() {
  return new Promise((resolve, reject) => {
    return process.nextTick(() => resolve(successResult));
  });
}
// defining default behavior of the mock to return success.
export const upload = jest.fn().mockImplementation(resolveSuccessfully);

export default {
  init: jest.fn(() => ({
    upload,
  })),
};
