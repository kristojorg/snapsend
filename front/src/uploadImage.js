// @flow
import React from 'react';
import type { Node } from 'react';
import filestack from 'filestack-js';
import accept from 'attr-accept';
import DropzoneComp from 'react-dropzone';
import type {
  Image as ImageType,
  UnfinishedEnvelope,
  Format,
  Size,
} from './types';

const fs = filestack.init(process.env.REACT_APP_FILESTACK_SECRET);

const SUCCESS_STATUS = 'Stored';
const LOGGED_OUT_LIMIT = 40;
const LOGGED_IN_LIMIT = 1000;

const WRONG_TYPE_ERROR = 'Input is not the right format';

export const ACCEPTED_TYPES = [
  'image/jpg',
  'image/jpeg',
  'image/pjpeg',
  'image/png',
  'image/webp',
  'image/tiff',
  'image/x-tiff',
];

export const FAILED_MESSAGE = 'Image Upload Failed';

export const handleDrop = (files: Array<File>, isLoggedIn: boolean) => {
  const overSignedOut = !isLoggedIn && files.length > LOGGED_OUT_LIMIT;
  const overSignedIn = isLoggedIn && files.length > LOGGED_IN_LIMIT;
  if (overSignedIn || overSignedOut) return 'Too many files';
  return files.map(file => uploadImage(file));
};

export const uploadImage = (file: File): Promise<ImageType> =>
  new Promise((resolve, reject) => {
    // make sure you got the right file type
    if (!accept(file, ACCEPTED_TYPES)) return reject(WRONG_TYPE_ERROR);

    fs
      .upload(file, {}, {})
      .then(res => {
        const { status } = res;
        if (status === SUCCESS_STATUS) {
          return resolve(res);
        }
        return reject(FAILED_MESSAGE);
      })
      .catch(err => reject(FAILED_MESSAGE));
  });

const Dropzone = ({
  children,
  onDrop,
  ...props
}: {
  children: Node,
  onDrop: (acceptedFiles: Array<File>) => void,
  props?: {},
}) => (
  <DropzoneComp
    onDrop={onDrop}
    accept={ACCEPTED_TYPES.join(',')}
    disableClick
    style={{
      position: 'relative',
      flex: 1,
      display: 'flex',
      flexDirection: 'column',
    }}
    {...props}
  >
    {children}
  </DropzoneComp>
);

export default Dropzone;

// export const transformMany = async (
//   images: Array<ImageType>,
//   maxWidth: ?number,
//   maxHeight: ?number
// ) => {
//   // do a promise all for each image.
//   const promiseArray = images.map(im => transform(im.url, maxWidth, maxHeight));
//   const res = await Promise.all(promiseArray);
//   return res;
// };

// export const transform = async (
//   handle: string,
//   maxWidth: ?number,
//   maxHeight: ?number
// ) => {
//   return fs.transform('url', {
//     resize: {
//       width: maxWidth,
//       height: maxHeight,
//     },
//   });
// };

// export const generateDownloadUrl = async (
//   images: Array<ImageType>,
//   envelope: ?UnfinishedEnvelope,
//   format: Format,
//   size: Size,
//   noneSelected: boolean
// ): Promise<string> => {
//   // start with the base url
//   const baseUrl = 'https://process.filestackapi.com/';

//   const transformedImages = await transformMany(
//     images,
//     size.width,
//     size.height
//   );

//   let imagesString = images.reduce((currString, im, ind) => {
//     const split = im.url.split('/');
//     const handle = split[split.length - 1];
//     if (im.selected || noneSelected) {
//       if (currString.length > 0) return `${currString},${handle}`;
//       return handle;
//     }
//     return currString;
//   }, '');

//   if (images.length > 1) imagesString = `[${imagesString}]`;

//   let formatString = '';
//   if (format === 'JPG') {
//     formatString = 'output=format:jpg/';
//   } else if (format === 'PNG') {
//     formatString = 'output=format:png/';
//   }

//   let resize = '';
//   if (size.width || size.height) {
//     let w = '';
//     let h = '';
//     if (size.width) w = `width:${size.width}`;
//     if (size.height) h = `height:${size.height}`;
//     if (size.width && size.height) {
//       w = `${w},`;
//       h = `${h}/`;
//     } else {
//       w = `${w}/`;
//     }

//     resize = `resize=${w}${h}`;
//   }
//   const result = `${baseUrl}${resize}${formatString}zip/${imagesString}`;
//   return result;
// };
