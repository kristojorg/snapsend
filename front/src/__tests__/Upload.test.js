jest.mock('filestack-js');
import filestack, { upload } from 'filestack-js';
import uploadImage from '../uploadImage';
import { successResult, errorResult } from './__mocks__/filestack-js';

describe('upload.js', () => {
  test('if upload succeeds, it should return the url', () => {
    uploadImage('image').then(res => {
      expect(result).toBe(successResult);
    });
  });

  test('When image upload fails, it returns the error code', () => {
    // somethinog
    uploadImage.mockImplementationOnce(() => Promise.reject(errorResult));
    uploadImage('image').then(res => {
      expect(result).toBe(errorResult);
    });
  });

  test('When input is not a file, it returns ___ error', () => {
    // something
  });

  test('When image is wrong format, it returns ___ error', () => {
    // something
  });
});

// function uploadImage(src){
// 	uploadingImage = true;
// 	const res = filestack.upload(src);

// 	if (error) {
// 		return "There was an error";
// 	}
// 	else if (res === 'undefined'){
// 		return "Upload failed";
// 	}
// 	else if (status === "OK"){
// 		return "Upload succeeded"
// 	}
// 	return "Upload failed";
// }

test('placeholder test', () => {});
