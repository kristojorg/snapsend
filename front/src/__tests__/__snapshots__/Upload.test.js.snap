// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handle files dropped on screen array with single incorrect format file 1`] = `"Input is not the right format"`;

exports[`Handle files dropped on screen multiple images, with one wrong 1`] = `"Input is not the right format"`;

exports[`upload to filestack When image is wrong format, it returns the right error 1`] = `"Input is not the right format"`;

exports[`upload to filestack When input is not a file, it returns the right error 1`] = `"Input is not the right format"`;

exports[`upload to filestack When upload fails, it returns the right error 1`] = `"Image Upload Failed"`;
